apps/examples/fs_test/smartfs_test.c

```diff
@@ -0,0 +1,331 @@
+{
+	int fd;
+	int ret;
+	int total_read = 0;
+	long time;
+	long timediff;
+
+	fd = open(filepath, O_RDONLY);
+	if (fd < 0) {
+		printf("%s errno: %d\n", __FUNCTION__, errno);
+		return ERROR;
+	}
+
+	time = get_time();
+	while (total_read < nbytes) {
+		ret = read(fd, buf + total_read, nbytes - total_read);
+		if (ret != nbytes - total_read) {
+			printf("%s errno: %d\n", __FUNCTION__, errno);
+			close(fd);
+			return ERROR;
+		}
+		total_read += ret;
+	}
+	timediff = get_time() - time;
+	g_total_time += timediff;
+	print_log("read:%ldms, ", timediff);
+
+	close(fd);
+
+	return OK;
+}
+
+static int fs_remove(char *filepath, void *buf, int nbytes)
+{
+	int ret;
+	long time;
+	long timediff;
+
+	time = get_time();
+	ret = unlink(filepath);
+	if (ret < 0) {
+		printf("%s errno: %d\n", __FUNCTION__, errno);
+		return ERROR;
+	}
+	timediff = get_time() - time;
+	g_total_time += timediff;
+	print_log("remove:%dms, ", timediff);
+
+	return OK;
+}
+
+static int write_read_test(char *filepath, void *buf, int nbytes, int iteration)
+{
+	int i;
+	int ret;
+
+	for (i = 0; i < iteration; i++) {
+		ret = fs_write(filepath, buf, nbytes);
+		if (ret < 0) {
+			return ERROR;
+		}
+
+		ret = fs_read(filepath, buf, nbytes);
+		if (ret < 0) {
+			return ERROR;
+		}
+	}
+
+	return OK;
+}
+
+static int write_test(char *filepath, void *buf, int nbytes, int iteration)
+{
+	int i;
+	int ret; 
+
+	for (i = 0; i < iteration; i++) {
+		ret = fs_write(filepath, buf, nbytes);
+		if (ret < 0) {
+			return ERROR;
+		}
+	}
+
+	return OK;
+}
+
+static int read_test(char *filepath, void *buf, int nbytes, int iteration)
+{
+	int i;
+	int ret;
+
+	ret = fs_write(filepath, buf, nbytes);
+	if (ret < 0) {
+		return ERROR;
+	}
+
+	for (i = 0; i < iteration; i++) {
+		ret = fs_read(filepath, buf, nbytes);
+		if (ret < 0) {
+			return ERROR;
+		}
+	}
+
+	return OK;
+}
+
+static int write_remove_test(char *filepath, void *buf, int nbytes, int iteration)
+{
+	int i;
+	int ret;
+
+	for (i = 0; i < iteration; i++) {
+		ret = fs_write(filepath, buf, nbytes);
+		if (ret < 0) {
+			return ERROR;
+		}
+
+		ret = fs_remove(filepath, buf, nbytes);
+		if (ret < 0) {
+			return ERROR;
+		}
+	}
+
+	return OK;
+}
+
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int smartfs_test_main(int argc, char *argv[])
+#endif
+{
+	int ret;
+	int size;
+	int operation;
+	int iteration;
+	char *buf;
+	char *filepath;
+
+	if (argc == 6) {
+		printf("FS R/W Performance Test!\n");
+
+		operation = atoi(argv[1]);
+		filepath = argv[2];
+		size = atoi(argv[3]);
+		iteration = atoi(argv[4]);
+		g_printopt = argv[5][0];
+
+		if (operation < TEST_OPT_WRITEREAD || operation >= TEST_OPT_MAX ||
+			size < 0 || iteration < 1 || iteration > FS_RW_ITERATION_MAX) {
+			printf("%s", USAGE);
+			return ERROR;
+		}
+
+		if (strlen(argv[5]) != 1 || (g_printopt != 'y' && g_printopt != 'n')) {
+			printf("%s", USAGE);
+			return ERROR;
+		}
+
+		buf = (char *)malloc(size);
```

Comment:
How about using `zalloc` instead of `malloc`?
Then we don't need to call `memset` in line 302.

Reply:
Good opinion. I modified that things.